pkg/apiserver/domain/service/cloudshell.go

```diff
@@ -205,6 +205,7 @@ func (c *cloudShellServiceImpl) prepareKubeConfig(ctx context.Context) error {
 			readOnly = checkReadOnly(p.Name, permissions)
 		}
 		if readOnly {
+			// TODO:(@barnettZQG) there needs a method to revoke the privileges
```

Comment:
revoke privileges for what?

Reply:
When users are deleted from a project or become viewers,  this change does not sync to the cluster role and role binding. Means permission leak in CloudShell.